flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -582,7 +583,18 @@ private void startTriggeringCheckpoint(CheckpointTriggerRequest request) {
 
 							return null;
 						},
-						timer));
+						timer)
+					.exceptionally(error -> {
+						isTriggering = false;
```

Comment:
I think we don't need to reset the internal state of `CheckpointCoordinator` here because it either is shut down or it will fail.